javascript-review.js

```diff
@@ -3,112 +3,203 @@
 
 // 14.
 
 // Define a function called numMult that takes two number arguments and a callback. The function will multiply the two numbers and pass the result to the callback as an argument. Now call numMult and in the callback simply console.log out the argument that has been passed through.
+const numMult = (num1, num2, displayCallback) => {
+  result = num1 * num2;
+  displayCallback(result);
+}
+
+numMult(3, 3, display);
 
 // 15.
 
 // a.
-// Define a function called addNum with that takes two number arguments. In this function simply return the addition of these numbers. 
+// Define a function called addNum with that takes two number arguments. In this function simply return the addition of these numbers.
+
+const addNum = (num1, num2) => {
+  return num1 + num2
+}
 
 // b.
 // Write a second function called numsPlusFunct that takes three arguments, two numbers and a function. Inside numsPlusFunct call the function that is passed as an argument, and pass the two number arguments to this function. numsPlusFunct will return an object where the first property has the value returned from that function call, and the second property is a string. 
 
+const numPlusFunct = (num1, num2, callback) => {
+  return {
+    result: callback(num1, num2),
+    ranString: "Yahaha! You found me!"
+  }
+}
+
 // c.
 // You have now made two functions. Call the numsPlusFunct and pass addNum as the appropriate argument. 
+console.log(numPlusFunct(2, 2, addNum));
 
 // 16.
 
 // What is Express? What does it help us to do?
+// Express is a node package that simplfy the routing process. It allows us to specify the routes and the method related to that route.
 
 // 17.
 
-// What is the difference between synchronous and asynchronous code? Name some ways that JS deals with the issue of asynchronous code. 
+// What is the difference between synchronous and asynchronous code? Name some ways that JS deals with the issue of asynchronous code.
+
+// sychronous code will run the line by line from top to bottom and will not proceed if the previous line of code is not fully processed.
+// asychronous code will generate a new thread in the processor to handle the asychronous code. this allows the application to simultanously to finish the sychronous code without having to wait for the asychronous code to finish.
+// The catch of asynchronous code is that there's a lot of external factor that may cause it to fail, such as internet connection, corrupted file, bad API server, etc... To overcome this Javascript created Promise object that allows the developer to specify the sucess and failure scenario.
 
 // 18.
 
 // What is fetch and how does it relate to AJAX? Give an example of how you would use it. What does fetch return? Give a very basic example of fetch.
+// fetch is a pre-built function that allows application to send a request to an API through AJAX engine that returns a Promise object. Before fetch, developer had to use XMLHttpRequest and it does not use AJAX engine.
```

Comment:
you're correct but `XMLHTTP` is still AJAX https://www.w3schools.com/XML/ajax_intro.asp